### DISCUSSION TITLE
Filtering optional 'to-one' relations

### DISCUSSION THREAD

**Discussion by Harukisatoh** [prisma/prisma#10643] [Q&A] [answered]
Hey, I'm stuck in a problem at a work project and I can't find a way to solve it, I hope you can help me.

Let's say that I have this schema:
```prisma
model Entry {
  id              Int                 @id @default(autoincrement())
  description     String?             @db.VarChar
  projectId       Int?
  project         Project?            @relation(fields: [projectId], references: [id])
}

model Client {
  id          Int        @id @default(autoincrement())
  name        String     @db.VarChar
  projects    Project[]
}

model Project {
  id          Int      @id @default(autoincrement())
  name        String   @db.VarChar
  clientId    Int?

  client      Client?     @relation(fields: [clientId], references: [id])
}
```

What I want to do is a query to find Entries that doesn't have any clients. So I tried to do this way:

```typescript
const entries = await this.prismaService.Entry.findMany({
      where: {
          OR: [
            {
              projectId: {
                equals: null,
              },
            },
            {
              project: {
                clientId: {
                  equals: null,
                },
              },
            },
          ],
      },
      select: {
        id: true,
        description: true,
        projectId: true,
        project: {
          select: {
            id: true,
            name: true,
            color: true,
            client: {
              select: {
                id: true,
                name: true,
              },
            },
          },
        },
      },
    });
```

The problem here is that I can have entries with a project with client null, and I also can have entries without project and therefore without client. I want to return both from query, but this approach just doesn't work, it alway returns an empty array.

**Comment by arthurdiluz** [comment]
I don't know

**Comment by austincrim** [comment, marked as answer]
Hey @Harukisatoh 👋 !

I recreated your query and it seems to be working for me after I tweaked your schema a bit.

```prisma
model Entry {
  id          Int      @id @default(autoincrement())
  description String?  @db.VarChar
  projectId   Int?
  project     Project? @relation(fields: [projectId], references: [id])
}

model Client {
  id       Int       @id @default(autoincrement())
  name     String    @db.VarChar
  projects Project[]
}

model Project {
  id       Int    @id @default(autoincrement())
  name     String @db.VarChar
  clientId Int?

  client Client? @relation(fields: [clientId], references: [id])
  Entry  Entry[]
}
```

I added the `Entry Entry[]` line in the `Project` model.

Could you make sure you have that line and try your query again?

**Reply by Harukisatoh** [reply]
Hey @austincrim,

I've forgotten to put this line on the schema. Anyway, I've made this change and it worked, thanks a lot!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
